modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/cmake-lists.mustache

```diff
@@ -8,16 +8,27 @@
 #
 # NOTE: Auto generated by OpenAPI Generator (https://openapi-generator.tech).
 
-cmake_minimum_required (VERSION 3.1)
+cmake_minimum_required (VERSION 3.5)
 
-project({{{packageName}}})
+project({{{packageName}}} CXX)
 
 # Force -fPIC even if the project is configured for building a static library.
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-set(CXX_STANDARD_REQUIRED ON)
 
+set(CXX_STANDARD_REQUIRED ON)
 if(NOT CMAKE_CXX_STANDARD)
-    set(CMAKE_CXX_STANDARD 14)
+    if(DEFINED CMAKE_CXX20_STANDARD_COMPILE_OPTION OR
+       DEFINED CMAKE_CXX20_EXTENSION_COMPILE_OPTION)
+      set(CMAKE_CXX_STANDARD 20)
+    elseif(DEFINED CMAKE_CXX17_STANDARD_COMPILE_OPTION OR
+           DEFINED CMAKE_CXX17_EXTENSION_COMPILE_OPTION)
+      set(CMAKE_CXX_STANDARD 17)
+    elseif(DEFINED CMAKE_CXX14_STANDARD_COMPILE_OPTION OR
+           DEFINED CMAKE_CXX14_EXTENSION_COMPILE_OPTION)
+      set(CMAKE_CXX_STANDARD 14)
+    else()
+      set(CMAKE_CXX_STANDARD 11)
+    endif()
 endif()
```

Comment:
Let's add some logging to be able to see what is happening. The test would need re-generation.


```suggestion
endif()
message(STATUS "Using C++ standard ${CMAKE_CXX_STANDARD}")
```

Reply:
looks like it's using C++ 11 according to the error message:
```
In file included from /home/circleci/OpenAPITools/openapi-generator/samples/client/petstore/cpp-restsdk/client/include/CppRestPetstoreClient/model/CreateUserOrPet_request.h:39:
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/variant:116:20: error: no template named 'add_const_t'; did you mean '::std::add_const_t'?
```

Reply:
That's the `libstdc++` version. But the compiler standard itself could be different.

Reply:
https://github.com/OpenAPITools/openapi-generator/pull/18821#discussion_r1628172625